src/main/java/com/algorithmia/algorithmHandler/SerializableException.java

```diff
@@ -0,0 +1,30 @@
+package com.algorithmia.algorithmHandler;
+
+import com.google.gson.JsonObject;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Arrays;
+
+class SerializableException<T extends  Throwable> {
```

Comment:
Making this class an extension of the `RuntimeException` hierarchy seems like the better way to go. That way, you get the message and stack trace for free. You can add the errorType as a new instance variable if you wish.
```
public class AlgorithmiaException extends RuntimeException {
  private String errorType;

 ...

  // then put your constructors which delegate to the base constructors for RuntimeException
}
```

Reply:
Agreed this might be cleaner, will explore this.